src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -643,11 +643,13 @@ private function createAuthenticationListeners(ContainerBuilder $container, stri
         }
 
         if ($container->hasDefinition('debug.security.firewall')) {
-            foreach ($authenticationProviders as $authenticatorId) {
-                $container->register('debug.'.$authenticatorId, TraceableAuthenticator::class)
-                    ->setDecoratedService($authenticatorId)
-                    ->setArguments([new Reference('debug.'.$authenticatorId.'.inner')])
+            foreach ($authenticationProviders as &$authenticatorId) {
+                $traceableId = 'debug.'.$authenticatorId;
+                $container
+                    ->register($traceableId, TraceableAuthenticator::class)
+                    ->setArguments([new Reference($authenticatorId)])
                 ;
+                $authenticatorId = $traceableId;
             }
```

Comment:
Is you code different from this simpler one (at least, your new test pass with my code as well)?

```suggestion
            foreach ($authenticationProviders as $authenticatorId) {
                $container
                    ->register('debug.'.$authenticatorId, TraceableAuthenticator::class)
                    ->setArguments([new Reference($authenticatorId)])
                ;
            }
```

Reply:
Yes: `$authenticationProviders`’ IDs wouldn’t be replaced by their traceable one in that case.

Reply:
Could you add an additional assertion that would fail with the suggested code? I guess something ensuring that the traceable service gets injected where it belongs

Reply:
I’m not sure it is possible: to make security-http 7.2 work with security-bundle < 7.2 the `TraceableAuthenticatorManagerListener` still must be able to decorate authenticators itself :thinking:

Reply:
Okay I thought this would fail with high or low-deps tests but it seems I’m still not grasming those :sweat_smile: 

Test case was refactored in https://github.com/symfony/symfony/pull/59278/commits/111913ed1507e89e6098f6bf82016570cdc5601a.